include/SH3/system/glvertarray.hpp

```diff
@@ -0,0 +1,249 @@
+/** @file
+ *
+ *  An abstraction of OpenGL Vertex Array Objects (VAO). Each mesh has its' own VAO, making it a lot simpler
+ *  to reuse meshes (for instance, just after Heather leaves the bathroom in the corridor with 3-4 Double Heads).
+ *
+ *  Also gives us a nice abstract and typesafe way to deal with VAOs.
+ *
+ *  @copyright 2017 Palm Studios
+ *
+ *  @date 8-6-2017
+ *
+ *  @author Jesse Buhagiar
+ */
+#ifndef GLVERTARRAY_HPP_INCLUDED
+#define GLVERTARRAY_HPP_INCLUDED
+
+#include <vector>
+#include <array>
+
+#include "GL/glew.h"
+#include "GL/gl.h"
+
+#include "SH3/system/assert.hpp"
+#include "SH3/system/glbuffer.hpp"
+
+namespace sh3_gl{
+
+    /**
+     *  Contains actual functionality of our VAO (@ref Bind, @ref Create etc).
+     *
+     *  @note Cannot be directly instantiated, use @ref vao instead
```

Comment:
Can != should. This should not be directly instantiated.
Also, add a trailing dot.